GiwazipClient/ViewModels/NetworkManager.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  NetworkManager.swift
+//  GiwazipClient
+//
+//  Created by 지준용 on 2023/02/07.
+//
+
+import Foundation
+
+import Alamofire
+import RxSwift
+import RxCocoa
+
+class NetworkManager {
+    static let shared = NetworkManager()
+    let header: HTTPHeaders = [APIEnvironment.apiField: APIEnvironment.apiKey]
+    var userData: User?
+    var roomData: Room?
+    
+    init() {
+        // TODO: 그때그때 방 번호가 달라져야 함!
+        loadUserData(url: APIEnvironment.usersURL + "/1")
+        loadUserData(url: APIEnvironment.usersURL + "/room/1")
+        
+        loadRoomData(url: APIEnvironment.roomsURL + "/1")
+        loadRoomData(url: APIEnvironment.roomsURL + "/category/1")
+        loadRoomData(url: APIEnvironment.roomsURL + "/post/1")
+        loadRoomData(url: APIEnvironment.roomsURL + "/post-category/1")
+        loadRoomData(url: APIEnvironment.roomsURL + "/user/1")
+    }
+
+    func requestUserData(url: String) -> Observable<NetworkResult<Any>> {
+        return Observable.create() { observer in
+            AF.request(url,
+                       method: .get,
+                       encoding: JSONEncoding.default,
+                       headers: self.header).responseData { (response) in
+                switch response.result {
+                case .success:
+                    guard let data = response.data,
+                          let statusCode = response.response?.statusCode
+                    else { return }
+                    let networkResult = self.judgeStatus(by: statusCode,
+                                                         self.isValidUserData(data: data))
+                    observer.onNext(networkResult)
+                    observer.onCompleted()
+                case .failure: break }
+            }
```

Comment:
p4;
해당 부분 indentation이 약간의 혼란을 야기하는 것 같습니다.
아래와 같이 변경되면 어떨까요?
```suggestion
            AF
                .request(url,
                         method: .get,
                         encoding: JSONEncoding.default,
                         headers: self.header)
                .responseData { response in
                    switch response.result {
                    case .success:
                        guard let data = response.data,
                              let statusCode = response.response?.statusCode
                        else { return }

                        let networkResult = self.judgeStatus(by: statusCode,
                                                             self.isValidUserData(data: data))
                        observer.onNext(networkResult)
                        observer.onCompleted()
                    case .failure:
                        break
                    }
```

Reply:
> p4; 해당 부분 indentation이 약간의 혼란을 야기하는 것 같습니다. 아래와 같이 변경되면 어떨까요?

36f1287 붙여쓰는게 더 깔끔해보여서 들여쓰기를 확인 못했네요

Reply:
> > p4; 해당 부분 indentation이 약간의 혼란을 야기하는 것 같습니다. 아래와 같이 변경되면 어떨까요?
> 
> [36f1287](https://github.com/Samsamhada/Giwazip-Client-App/commit/36f1287163010568c9999857ae3127a0d0a4e892) 붙여쓰는게 더 깔끔해보여서 들여쓰기를 확인 못했네요

사실 여기 들여쓰기의 핵심은 chaining 되고 있는 responseData가 눈에 안들어와서 그 부분의 수정이 우선이었는데, 해당 부분은 수정이 안되어있네요..

Reply:
> > > p4; 해당 부분 indentation이 약간의 혼란을 야기하는 것 같습니다. 아래와 같이 변경되면 어떨까요?
> > 
> > 
> > [36f1287](https://github.com/Samsamhada/Giwazip-Client-App/commit/36f1287163010568c9999857ae3127a0d0a4e892) 붙여쓰는게 더 깔끔해보여서 들여쓰기를 확인 못했네요
> 
> 사실 여기 들여쓰기의 핵심은 chaining 되고 있는 responseData가 눈에 안들어와서 그 부분의 수정이 우선이었는데, 해당 부분은 수정이 안되어있네요..

77aad40 인덴트가 컨벤션을 지키지 않아 수정하지 않았으나, 가독성면에서는 더 좋아보여 수정했습니다.

Reply:
> > > > p4; 해당 부분 indentation이 약간의 혼란을 야기하는 것 같습니다. 아래와 같이 변경되면 어떨까요?
> > > 
> > > 
> > > [36f1287](https://github.com/Samsamhada/Giwazip-Client-App/commit/36f1287163010568c9999857ae3127a0d0a4e892) 붙여쓰는게 더 깔끔해보여서 들여쓰기를 확인 못했네요
> > 
> > 
> > 사실 여기 들여쓰기의 핵심은 chaining 되고 있는 responseData가 눈에 안들어와서 그 부분의 수정이 우선이었는데, 해당 부분은 수정이 안되어있네요..
> 
> [77aad40](https://github.com/Samsamhada/Giwazip-Client-App/commit/77aad4073073afcfdf937b3c49a30937e1cfe262) 인덴트가 컨벤션을 지키지 않아 수정하지 않았으나, 가독성면에서는 더 좋아보여 수정했습니다.

확인했습니다. 리뷰 리졸브합니다~